scripts/utils/verify/writeInfo.ts

```diff
@@ -0,0 +1,87 @@
+import { upgrades } from "hardhat";
+
+
+import { SECTION_SEPARATOR, SUBSECTION_SEPARATOR } from "../bootstrap-tools";
+
+import { deploymentData, PromissoryNoteTypeBn, PromissoryNoteTypeLn } from "../../deploy/deploy";
+
+export let contractInfo: deploymentData = {}
+export async function main (
+    assetVaultAddress: string,
+    feeControllerAddress: string,
+    borrowerNoteAddress: string,
+    lenderNoteAddress: string,
+    repaymentContAddress: string,
+    whitelistAddress: string,
+    vaultFactoryAddress: string,
+    loanCoreAddress: string,
+    originationContAddress: string
+): Promise<void> {
+    contractInfo["CallWhitelist"] = {
+        "contractAddress": whitelistAddress,
+
+        "constructorArgs": []
+    };
+
+     contractInfo["AssetVault"] = {
+        "contractAddress": assetVaultAddress,
+
+        "constructorArgs": []
+    };
+
+    const factoryProxyAddress = vaultFactoryAddress
+    const factoryImplAddress = await upgrades.erc1967.getImplementationAddress(factoryProxyAddress)
+    contractInfo["VaultFactory"] = {
+        "contractAddress": factoryImplAddress,
+
+        "constructorArgs": []
+    };
+
+    contractInfo["FeeController"] = {
+        "contractAddress": feeControllerAddress,
+
+        "constructorArgs": []
+    };
+
+    let promissoryNoteDataBn: PromissoryNoteTypeBn = {
+
+        "contractAddress": borrowerNoteAddress,
+
+        "constructorArgs": ["Arcade.xyz BorrowerNote", "aBN"]
```

Comment:
I think these should not be hardcoded, but should be synced up with what's in `deploy.ts` - they should be passed via the call to `writeJson`, so we keep things DRY and make sure our configured constructor args exist only in one place.

Going even further, it might be better to actually have the constructor args source of truth be _native_ to a JSON file, and have `deploy.ts` import the JSON file to figure out what args it needs to use for each deploy. Then, we can avoid passing them as function parameters, and `writeJson` can simply import the same constructor args from the separate JSON file.

Reply:
100% agreed.

Reply:
Refactored to pass the lenderNote and borrowerNote args from deploy.ts into writeJson.

Re. the constructor args being native to a JSON file, as discussed,  leaving them as is since they are generated by deploy.ts and used within it.